### PR TITLE
Replace noop Errorf calls with Printf calls

### PR DESCRIPTION
**Pull request by whyrusleeping**
Was debugging some stream muxing issues when i noticed these errors werent being properly logged.

### CODE REVIEW

**Comment by whyrusleeping** [comment]
@dmcgowan I also added a fix for the bug i was debugging. The priority specified in the SynStreamFrame was never getting set to the stream object, causing the priorities of packets in the priority queue getting mixed up, leading to data frames getting handled before their synstream frame. Which ended up with us losing data.

File: connection.go [hunk 1]
```diff
@@ -421,7 +422,7 @@ func (s *Connection) frameHandler(frameQueue *PriorityFrameQueue, newHandler Str
 		}
 
 		if frameErr != nil {
-			fmt.Errorf("frame handling error: %s", frameErr)
+			log.Printf("frame handling error: %s\n", frameErr)
```
Comment (dmcgowan) [inline connection.go]: nit: the new line isn't necessary

**Comment by dmcgowan** [comment]
Thanks for this change. Clearly this project needs CI setup to run `go vet`!

**Comment by dmcgowan** [comment]
Do you think you could update to use `debugMessage` instead?

**Comment by whyrusleeping** [comment]
@dmcgowan sure thing!

File: connection.go (hunk 1 above)
Comment (whyrusleeping) [inline connection.go]: sorry about that, initially had it as a `fmt.Printf` and missed removing it

**Comment by whyrusleeping** [comment]
@dmcgowan Updated!

**Comment by dmcgowan** [comment]
LGTM

**Pull request merged**